lib/manager/regex/index.ts

```diff
@@ -108,18 +113,47 @@ function handleAny(
   return config.matchStrings
     .map((matchString) => regEx(matchString, 'g'))
     .flatMap((regex) => regexMatchAll(regex, content)) // match all regex to content, get all matches, reduce to single array
-    .map((matchResult) => createDependency(matchResult, config));
+    .map((matchResult) => createDependency(matchResult, null, config));
+}
+
+function mergeGroups(
```

Comment:
Can you make sure there's no possibility for prototype pollution this way? :-/

Reply:
I check now if the match group is a valid match field